store/store.go

```diff
@@ -1178,6 +1198,49 @@ var download = func(name, downloadURL string, user *auth.UserState, s *Store, w
 	return err
 }
 
+// downloadDelta downloads the delta for the preferred format, returning the path.
+func (s *Store) downloadDelta(name string, downloadDir string, downloadInfo *snap.DownloadInfo, pbar progress.Meter, user *auth.UserState) (string, error) {
+
+	if len(downloadInfo.Deltas) != 1 {
+		return "", errors.New("store returned more than one delta")
+	}
+
+	deltaInfo := downloadInfo.Deltas[0]
+
+	if deltaInfo.Format != s.deltaFormat {
+		return "", fmt.Errorf("store returned delta with format %q", deltaInfo.Format)
```

Comment:
if I saw this error I wouldn't know what the *expected* delta format was without digging a lot more